packages/arb-util/wsbroadcastserver/clientmanager.go

```diff
@@ -144,65 +146,59 @@ func (cm *ClientManager) Broadcast(bm interface{}) {
 	cm.broadcastChan <- bm
 }
 
-func (cm *ClientManager) doBroadcast(bm interface{}) error {
+func (cm *ClientManager) doBroadcast(bm interface{}) ([]*ClientConnection, error) {
 	if err := cm.catchupBuffer.OnDoBroadcast(bm); err != nil {
-		return err
+		return nil, err
 	}
 
 	var buf bytes.Buffer
 	writer := wsutil.NewWriter(&buf, ws.StateServerSide, ws.OpText)
 	encoder := json.NewEncoder(writer)
 	if err := encoder.Encode(bm); err != nil {
-		return errors.Wrap(err, "unable to encode message")
+		return nil, errors.Wrap(err, "unable to encode message")
 	}
 	if err := writer.Flush(); err != nil {
-		return errors.Wrap(err, "unable to flush message")
+		return nil, errors.Wrap(err, "unable to flush message")
 	}
 
 	clientDeleteList := make([]*ClientConnection, 0, len(cm.clientPtrMap))
 	for client := range cm.clientPtrMap {
-		if len(client.out) == MaxSendQueue {
-			// Queue for client too backed up, so delete after going through all other clients
+		if len(client.out) >= cm.maxSendQueue {
+			// Queue for client too backed up, disconnect instead of blocking on channel send
+			logger.Error().Str("client", client.Name).Int("sendQueue", len(client.out)).Msg("disconnecting because sendQueue too large")
```

Comment:
Let's make this an info line to avoid spamming the logs. It's the client's fault in this case.